app/models/core_data_connector/item.rb

```diff
@@ -9,7 +11,65 @@ class Item < ApplicationRecord
     include UserDefinedFields::Fieldable
     include Search::Item
 
+    after_create :fcc_import
+
+    CODE_NO_RESPONSE = 0
+
     # Nameable table
     name_table :source_titles, polymorphic: true
+
+    # Generate the full URL for the item's CSV files in FairCopy.cloud
+    def faircopy_cloud_url
+      if !self[:faircopy_cloud_id]
+        return nil
+      end
+
+      project = self.project_model.project
+
+      if !project[:faircopy_cloud_url]
+        return nil
+      end
+
+      "#{project[:faircopy_cloud_url]}/documents/#{self[:faircopy_cloud_id]}/csv"
+    end
+
+    def fetch_csv_zip
+      url = self.faircopy_cloud_url
+
+      if !url
+        return nil
+      end
+
+      request = Typhoeus::Request.new(url, followlocation: true)
```

Comment:
This looks very similar to what we're doing in `/services/core_data_connector/authority/http.rb`. Could we extract any of that to a reusable location?

Reply:
Would creating a new service called `Http` make sense for reusing the `http_request` function?

In JS I would just extract it into something like `http_helper.js` and import the function in both places. Rails seems to make reusing single functions like this really tricky unless you're working within a specific scope (model or controller concerns) because of the rules about what's being imported, etc. Or maybe the fact that OOP doesn't really expect you to have generic no-side-effects helper functions outside of a class?

So services are the only way I figured out how to reuse code between models and controllers when working on the `ZipHelper` code. It seems like some people use a `lib` folder for sharing functions but it requires config changes to autoload it?

Reply:
Yeah, I think an HTTP service is probably the way to go. It can house anything that uses `typhoeus`. Maybe something like:

```
├── services
│   ├── core_data_connector
│   │   ├── http
|   │   │   ├── requestable.rb
```